nearcore/src/test_utils.rs

```diff
@@ -22,35 +25,31 @@ impl TestEnvNightshadeSetupExt for TestEnvBuilder {
         self.nightshade_runtimes_with_runtime_config_store(genesis, runtime_configs)
     }
 
-    fn nightshade_runtimes_with_runtime_config_store(
+    fn nightshade_runtimes_with_runtime_config_store<'a>(
         self,
-        genesis: &Genesis,
+        genesis: &'a Genesis,
         runtime_configs: Vec<RuntimeConfigStore>,
     ) -> Self {
-        let (builder, home_dirs, stores, epoch_managers, state_snapshot_enabled) =
-            self.internal_ensure_epoch_managers_for_nightshade_runtime();
-        assert_eq!(runtime_configs.len(), epoch_managers.len());
-        let runtimes = stores
-            .into_iter()
-            .zip(home_dirs)
-            .zip(epoch_managers)
-            .zip(runtime_configs)
-            .map(|(((store, home_dir), epoch_manager), runtime_config)| {
-                // TODO: It's not ideal to initialize genesis state with the nightshade runtime here for tests
-                // Tests that don't use nightshade runtime have genesis initialized in kv_runtime.
-                // We should instead try to do this while configuring store.
-                let home_dir = home_dir.as_path();
-                initialize_genesis_state(store.clone(), genesis, Some(home_dir));
-                NightshadeRuntime::test_with_runtime_config_store(
-                    home_dir,
-                    store,
-                    &genesis.config,
-                    epoch_manager,
-                    runtime_config,
-                    state_snapshot_enabled,
-                ) as Arc<dyn RuntimeAdapter>
-            })
-            .collect();
-        builder.runtimes(runtimes)
+        let state_snapshot_enabled = self.state_snapshot_enabled();
+        let nightshade_runtime_creator = |home_dir: PathBuf,
```

Comment:
The reason why we have this function here and not as part of the original test_util is that NightshadeRuntime is only a part of nearcore module and we don't really have access to it in client module.

Keeping that in mind, it makes more sense to just pass a "nightshade runtime creator/constructor" function to the TestEnvBuilder